### PR TITLE
K8SPS-73 - Add haproxy test

### PR DESCRIPTION
**Pull request by tplavcic**
[K8SPS-73](https://jira.percona.com/browse/K8SPS-73) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/demand-backup.log) |
| auto-config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/monitoring.log) |
| semi-sync | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/semi-sync.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/service-per-pod.log) |
| sidecars | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/sidecars.log) |
| limits | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/limits.log) |
| version-service | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/version-service.log) |
| tls-cert-manager | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/tls-cert-manager.log) |
| config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/config.log) |
| gr-demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/gr-demand-backup.log) |
| one-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/one-pod.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/scaling.log) |
| haproxy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/haproxy.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/cc17a4f/users.log) |
| We run 17 out of 17|

commit: https://github.com/percona/percona-server-mysql-operator/pull/249/commits/cc17a4f5b57b6c43162328cc86061db0cb9eb225
image: `perconalab/percona-server-mysql-operator:PR-249-cc17a4f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/semi-sync.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/config.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/service-per-pod.log) |
| gr-demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/gr-demand-backup.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/sidecars.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/one-pod.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/limits.log) |
| haproxy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/haproxy.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/version-service.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/0c9c2d4/users.log) |
| We run 17 out of 17|

commit: https://github.com/percona/percona-server-mysql-operator/pull/249/commits/0c9c2d460b7f88cab6e89586c7a10a3c24749d32
image: `perconalab/percona-server-mysql-operator:PR-249-0c9c2d4`

File: e2e-tests/tests/haproxy/07-check-primary-failover.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+kind: TestStep
+timeout: 30
+commands:
+  - script: |-
+      set -o errexit
+      set -o xtrace
+
+      source ../../functions
+
+      primary_label="$(get_primary_from_label)"
```
Comment (cap1984) [inline e2e-tests/tests/haproxy/07-check-primary-failover.yaml:11]: maybe `primary_pod_from_label`?

**Review by cap1984 — DISMISSED** [review]
LGTM

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/auto-config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/semi-sync.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/config.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/service-per-pod.log) |
| gr-demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/gr-demand-backup.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/limits.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/one-pod.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/version-service.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/haproxy.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/users.log) |
| We run 17 out of 17|

commit: https://github.com/percona/percona-server-mysql-operator/pull/249/commits/bd5527a64981941a2cffbbbbf2b4078ae1e395fc
image: `perconalab/percona-server-mysql-operator:PR-249-bd5527a`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/demand-backup.log) |
| gr-demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/gr-demand-backup.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/config.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/one-pod.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/init-deploy.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/haproxy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/limits.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/version-service.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/tls-cert-manager.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-249/bd5527a/users.log) |
| We run 17 out of 17|

commit: https://github.com/percona/percona-server-mysql-operator/pull/249/commits/bd5527a64981941a2cffbbbbf2b4078ae1e395fc
image: `perconalab/percona-server-mysql-operator:PR-249-bd5527a`

**Review by nmarukovich — DISMISSED** [review]

File: e2e-tests/tests/haproxy/07-check-primary-failover.yaml (hunk 1 above)
Comment (tplavcic) [reply e2e-tests/tests/haproxy/07-check-primary-failover.yaml]: Fixed: 8a2497f75b2c2e68a42172cf5baa94e7f62d3f57

**Pull request merged**